include/openmc/mesh.h

```diff
@@ -241,6 +241,7 @@ class StructuredMesh : public Mesh {
   xt::xtensor<double, 1> lower_left_;  //!< Lower-left coordinates of mesh
   xt::xtensor<double, 1> upper_right_; //!< Upper-right coordinates of mesh
   std::array<int, 3> shape_; //!< Number of mesh elements in each dimension
+  Position origin_ {0.0, 0.0, 0.0};
```

Comment:
Do we want this here? in the docs and python API it seems like we are only allowing `origin` for `CylindricalMesh` and `SphericalMesh`, but now all the `StructuredMesh` classes will have this member. As we are trying to make our Mesh functionality more consistent across the code base it might be a good time to clean this up too.

Reply:
Should this only be in `CylindricalMesh` and `SphericalMesh` then?

Reply:
I debated internally about this too. The problem I ran into with placing this in the `CylindricalMesh`/`SphericalMesh` classes is that the origin needs to be accounted for in the `StructuredMesh::raytrace_mesh` method. I think I have a better way to handle this now though. I'll push a commit later today.